laurent22/joplin
# Desktop,Mobile: Update Mermaid to v10.9.1

# Summary

This pull request upgrades `mermaid.js` to version 10.9.1 from 10.6.1. See [the mermaid releases page](https://github.com/mermaid-js/mermaid/releases) for a list of changes.

# Testing plan

1. Create a new note with the following content:
    ````
    ```mermaid
    graph LR
	    A[$$n$$]
	    B["$$n \over 2$$"]
	    C["$$\frac{n}{2}$$"]
	    A--"$$\mathcal{O}(1)$$"-->B
	    A--"$$\mathcal{O}(1)$$"-->C
    ```
    ````
2. Verify that the flowchart renders (and that the math also renders):
    ![screenshot: Flowchart with math](https://github.com/laurent22/joplin/assets/46334387/66a1d77f-727f-412b-ad5f-f66bb580cb47)
3. Desktop only:
   1. Download a copy of the diagram using the download button and attach it to the note.
   2. Switch to the rich text editor.
   3. Verify that the original diagram and the attached SVG export render.

This has been tested successfully on Ubuntu 24.04 (desktop) and on an Android 12 emulator (mobile).
